crates/binjs_es6/src/scopes.rs

```diff
@@ -826,24 +826,24 @@ impl Visitor<()> for AnnotationVisitor {
     }
 
     fn enter_function_expression_contents(&mut self, path: &Path, _node: &mut FunctionExpressionContents) -> Result<VisitMe<()>, ()> {
-        self.push_var_scope(path);
-        self.push_param_scope(path);
-        self.push_this_captured();
         if let Some(_) = self.function_expression_name() {
             self.push_function_name_captured();
         }
+        self.push_var_scope(path);
```

Comment:
Since this is obviously important (and since I wrote that code and even I can't remember why), could you add a comment explaining why this needs to happen after the `if let` above?